src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -889,6 +889,9 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
   gboolean is_uninstall = FALSE;
   gboolean is_override = FALSE;
 
+  if (deploy_has_bool_option (self, "apply-live") && deploy_has_bool_option (self, "reboot"))
+    return glnx_throw (error, "Cannot specify `apply-live` and `reboot`");
```

Comment:
This would belong better in `rpmostreed_transaction_new_deploy` I think alongside the other semantic checks we already do there, but meh not worth a respin!